util/file_splitter.go

```diff
@@ -66,6 +70,36 @@ func SplitFile(filePath string, fileChunkSizeInMB uint64) ([]string, error) {
 	return fileParts, nil
 }
 
+// ValidateChunkSize validate the chunk size
+func ValidateChunkSize(filePath string, fileChunkSizeInMB uint64) error {
+	if fileChunkSizeInMB == 0 {
+		return nil
+	}
+
+	if fileChunkSizeInMB == configuration.DefaultChunkSizeInMB {
+		return nil
+	}
+
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return err
+	}
+
+	var fileSize = uint64(fileInfo.Size())
+	var fileSizeInMb = fromBytes(fileSize)
+
+	var minFileChunkSizeInMb = uint64(math.Ceil(float64(fileSizeInMb) / float64(DefaultFileChunkCount)))
+
+	if fileChunkSizeInMB < minFileChunkSizeInMb {
+		return fmt.Errorf("The custom specified chunk size %d MB cannot be less than the minimum chunk size %d MB for archive with size %d MB", fileChunkSizeInMB, minFileChunkSizeInMb, fileSizeInMb)
+	}
+
+	if fileChunkSizeInMB > configuration.DefaultChunkSizeInMB {
```

Comment:
Why do we need this check? If we keep it, we also have to document it.

Reply:
Because if you upload 3GB archive fileChunkSizeInMB could exceed the 50MB which with slow connection could make the gorouter to drop the connection.